### DISCUSSION TITLE
Omitting TextArea theme doesn't highlight

### DISCUSSION THREAD

**Discussion by davidbrochart** [Textualize/textual#4148] [Q&A] [answered]
It seems that passing a `theme` to a `TextArea` is mandatory to get syntax highlighting, is it intended?
```py
from textual.app import App, ComposeResult
from textual.widgets import TextArea

TEXT = """\
def hello(name):
    print("hello" + name)

def goodbye(name):
    print("goodbye" + name)
"""


class TextAreaExample(App):
    def compose(self) -> ComposeResult:
        # doesn't highlight
        yield TextArea(TEXT, language="python")
        # does highlight
        yield TextArea(TEXT, language="python", theme="monokai")


app = TextAreaExample()
if __name__ == "__main__":
    app.run()
```
![image](https://github.com/Textualize/textual/assets/4711805/f4be2dc3-0127-4484-aed0-0d0b37919f1f)
# Textual Diagnostics

## Versions

| Name    | Value  |
|---------|--------|
| Textual | 0.50.1 |
| Rich    | 13.7.0 |

## Python

| Name           | Value                                                                     |
|----------------|---------------------------------------------------------------------------|
| Version        | 3.12.1                                                                    |
| Implementation | CPython                                                                   |
| Compiler       | GCC 12.3.0                                                                |
| Executable     | /home/david/.local/share/hatch/env/virtual/jpterm/-V8cnvgI/dev/bin/python |

## Operating System

| Name    | Value                                       |
|---------|---------------------------------------------|
| System  | Linux                                       |
| Release | 5.15.0-94-generic                           |
| Version | #104-Ubuntu SMP Tue Jan 9 15:25:40 UTC 2024 |

## Terminal

| Name                 | Value          |
|----------------------|----------------|
| Terminal Application | *Unknown*      |
| TERM                 | xterm-256color |
| COLORTERM            | truecolor      |
| FORCE_COLOR          | *Not set*      |
| NO_COLOR             | *Not set*      |

## Rich Console options

| Name           | Value                |
|----------------|----------------------|
| size           | width=136, height=35 |
| legacy_windows | False                |
| min_width      | 1                    |
| max_width      | 136                  |
| is_terminal    | True                 |
| encoding       | utf-8                |
| max_height     | 35                   |
| justify        | None                 |
| overflow       | None                 |
| no_wrap        | False                |
| highlight      | None                 |
| markup         | None                 |
| height         | None                 |

**Comment by davep** [comment]
If your intention is to create a `TextArea` that is intended for code editing, complete with syntax highlighting and the like, [you should probably be looking at using `TextArea.code_editor`](https://textual.textualize.io/widgets/text_area/#code-editing-vs-plain-text-editing).

Note that this approach is new [as of 0.48.0](https://github.com/Textualize/textual/releases/tag/v0.48.0).

**Reply by davidbrochart** [reply]
Thanks, yes that's what I saw, but still it feels weird that `TextArea` accepts a `language` parameter which is ignored if no `theme` is passed. Maybe there should be a runtime check that sets a default theme if a language was passed without a theme?

**Reply by davep** [reply]
If using `code_editor` to make a code-oriented editor (which would make sense, as there's a number of other things that aren't code-friendly about `TextArea` "out of the box") it will default the theme.

**Reply by darrenburns** [reply]
Yeah this was mentioned elsewhere and I agree. The default CSS theme just has no syntax highlighting mapping on it at the moment, but it probably should have at least the basic highlighting. The `language` parameter isn't ignored - by specifying a language it means tree-sitter is parsing the content behind the scenes and returning syntax tree nodes to us, but because the CSS theme doesn't have a mapping defined, you don't see any colours.

**Reply by darrenburns** [reply, marked as answer]
I've made this PR in relation to this: https://github.com/Textualize/textual/pull/4149

**Reply by davidbrochart** [reply]
Thanks!
